src/packages/core/src/server.ts

```diff
@@ -183,27 +183,24 @@ export class Server extends Emittery<{ open: undefined; close: undefined }> {
           else throw err;
         }
       })
-    ]).catch(async error => {
+    ]).then((promiseResults) => {
+      if (promiseResults[0].status === "fulfilled" && promiseResults[1].status === "fulfilled") {
+        this.emit("open");
+      } else {
+        let reason = "";
+        reason += promiseResults[0].status === "rejected" ? `${promiseResults[0].reason}\n\n` : "";
+        reason += promiseResults[1].status === "rejected" ? promiseResults[1].reason : "";
+        throw reason;
+      }
```

Comment:
Another possibility here is to only send the _first_ error we get back to the user. It'll be clearer to the user what the problem is (since `AggregateError`s aren't something developers read every day). Doing it this way (first error only) may solves the complexity of handling these errors.